src/Rule/Each.php

```diff
@@ -57,6 +57,7 @@ public function validate($value)
 
         $result = true;
         foreach ($value as $index => $innerValue) {
+            $innerValue['array_index'] = $index;
```

Comment:
Nice hack to be able to validate the array keys, I like it! The only thing that I'm thinking about is the key name. Do we want to go with an underscore here, or do we want to go with camelCase, like php variables should be?